repose-aggregator/components/services/datastore-service/datastore-service-api/src/main/resources/META-INF/schema/config/hazelcast-datastore.xsd

```diff
@@ -0,0 +1,222 @@
+                <xs:attribute name="auto-increment" type="xs:boolean" use="optional" default="true"/>
+                <xs:attribute name="port-count" type="xs:unsignedInt" use="optional" default="100"/>
+            </xs:extension>
+        </xs:simpleContent>
+    </xs:complexType>
+
+    <xs:complexType name="Join">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>
+                    The `join` configuration element is used to enable the Hazelcast instances to form a cluster,
+                    i.e. to join the members. The following ways can be used to join the members: discovery by TCP/IP,
+                    by multicast, by discovery on Kubernetes, and by discovery on AWS (EC2 auto-discovery).
+                </html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:choice>
+            <xs:element name="tcp-ip" type="Tcp-ip"/>
+            <xs:element name="multicast" type="Multicast"/>
+            <xs:element name="aws" type="Aliased-discovery-strategy"/>
+            <xs:element name="kubernetes" type="Aliased-discovery-strategy"/>
+        </xs:choice>
+    </xs:complexType>
+
+    <xs:complexType name="Tcp-ip">
+        <xs:sequence>
+            <xs:element name="member" type="xs:string" minOccurs="1" maxOccurs="unbounded" default="127.0.0.1">
+                <xs:annotation>
+                    <xs:documentation>
+                        <html:p>
+                            IP address(es) of one or more well known members. Once members are connected to these
+                            well known ones, all member addresses will be communicated with each other.
+                        </html:p>
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:element>
+        </xs:sequence>
+
+        <xs:attribute name="connection-timeout-seconds" type="xs:int" use="optional" default="5">
+            <xs:annotation>
+                <xs:documentation>
+                    <html:p>
+                        The maximum amount of time Hazelcast is going to try to connect to a well known member
+                        before giving up. Setting it to a too low value could mean that a member is not able
+                        to connect to a cluster. Setting it to a too high value means that member startup could
+                        slow down because of longer timeouts (e.g. when a well known member is not up). Increasing
+                        this value is recommended if you have many IPs listed and the members cannot properly
+                        build up the cluster. Its default value is 5.
+                    </html:p>
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+    </xs:complexType>
+
+    <xs:complexType name="Multicast">
+        <xs:all>
+            <xs:element name="multicast-group" type="xs:string" minOccurs="0" maxOccurs="1" default="224.2.2.3">
```

Comment:
These seem like they should be optional attributes instead of elements.

Reply:
While I would agree and can make that change, the reason they are `element`s is that is how they are defined in Hazelcast's configuration. That is, I left them as `element`s to make it as easy as possible to transition from our configuration to a full Hazelcast configuration.

Reply:
unfortunate but understandable.